scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/goals/CardinalityGoal.java

```diff
@@ -45,6 +49,19 @@ public class CardinalityGoal extends ActivityTemplateGoal {
    */
   protected TimeRangeExpression expr;
 
+  // Following members are related to diagnosis of the scheduler being stuck with inserting 0-duration activities
+  /**
+   * Activities inserted so far to satisfy this goal
+   */
+  private Map<SchedulingActivityInstanceId, ActivityInstance> insertedSoFar = new HashMap<>();
```

Comment:
Since we never retrieve the values in this Map, I think we only need a Set:

```suggestion
  private Map<SchedulingActivityInstanceId> insertedSoFar = new HashSet<>();
```